src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -1288,25 +1298,29 @@ private <T> void resetTaskStateAsStopped(
     }
 
     private void resetEntityTasksAsStopped(String detectorTaskId) {
-        UpdateByQueryRequest updateByQueryRequest = new UpdateByQueryRequest();
-        updateByQueryRequest.indices(DETECTION_STATE_INDEX);
-        BoolQueryBuilder query = new BoolQueryBuilder();
-        query.filter(new TermQueryBuilder(PARENT_TASK_ID_FIELD, detectorTaskId));
-        query.filter(new TermQueryBuilder(TASK_TYPE_FIELD, ADTaskType.HISTORICAL_HC_ENTITY.name()));
-        query.filter(new TermsQueryBuilder(STATE_FIELD, NOT_ENDED_STATES));
-        updateByQueryRequest.setQuery(query);
-        updateByQueryRequest.setRefresh(true);
+
         String script = String.format(Locale.ROOT, "ctx._source.%s='%s';", STATE_FIELD, ADTaskState.STOPPED.name());
-        updateByQueryRequest.setScript(new Script(script));
+        UpdateByQueryRequest updateByQueryRequest = new UpdateByQueryRequest.Builder().index(DETECTION_STATE_INDEX).query(queryBuilder -> {
+            queryBuilder.match(mb -> mb.field(PARENT_TASK_ID_FIELD).query(vb -> vb.stringValue(detectorTaskId)));
+            queryBuilder.match(mb -> mb.field(TASK_TYPE_FIELD).query(vb -> vb.stringValue(ADTaskType.HISTORICAL_HC_ENTITY.name())));
+            for (String notEndedStates : NOT_ENDED_STATES) {
+                queryBuilder.match(mb -> mb.field(STATE_FIELD).query(vb -> vb.stringValue(notEndedStates)));
+            }
```

Comment:
Can we use`matchAll` instead of a for loop here?

Reply:
I had to redo some of the UpdateByQueryRequests and DeleteByQueryRequests to use the correct builders. MatchAll doesnt apply anymore